source-code/sdk-js/src/adapter-sveltekit/vite-plugin/config/config.ts

```diff
@@ -0,0 +1,158 @@
+	if (transformConfig) return transformConfig
+
+	const nodeishFs = await getNodeishFs()
+
+	// eslint-disable-next-line no-async-promise-executor
+	return (transformConfig = new Promise<TransformConfig>(async (resolve, reject) => {
+		const inlang = await openInlangProject({ nodeishFs, configPath: PATH_TO_INLANG_CONFIG })
+
+		const errors = inlang.errors()
+		if (errors.length) {
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const error = errors[0]!
+			if (error instanceof ConfigPathNotFoundError) {
+				await createBasicInlangConfig(nodeishFs)
+				resetTransformConfig()
+				return resolve(initTransformConfig())
+			}
+
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			return reject(error)
+		}
+
+		await updateSdkModuleVersion(inlang)
+		await createDemoResourcesIfNoMessagesExistYet(inlang)
+
+		const settings = getSettings(inlang)
+		if (!settings) {
+			resetTransformConfig()
+			return resolve(initTransformConfig())
+		}
+		if (settings instanceof InlangSdkException) {
+			return reject(settings)
+		}
+
+		// we can't use `import` in tests to import something from the virtual file system
+		// so we need to use `createImport` instead
+		const { default: svelteConfig } = (await (import.meta.env?.TEST
+			? createImport({ readFile: nodeishFs.readFile, fetch })(PATH_TO_SVELTE_CONFIG)
+			: import(/* @vite-ignore */ PATH_TO_SVELTE_CONFIG)
+		).catch((error: unknown) => {
+			return {
+				default: new InlangSdkException(
+					`Could not find 'svelte.config.js' file (${PATH_TO_SVELTE_CONFIG})`,
+					error as Error,
+				),
+			}
+		})) as { default: SvelteConfig | InlangSdkException }
+		if (svelteConfig instanceof InlangSdkException) {
+			return reject(svelteConfig)
+		}
+
+		const files = {
+			appTemplate: path.resolve(
+				PATH_TO_CWD,
+				svelteConfig.kit?.files?.appTemplate || path.resolve("src", "app.html"),
+			),
+			routes: path.resolve(
+				PATH_TO_CWD,
+				svelteConfig.kit?.files?.routes || path.resolve("src", "routes"),
+			),
+			serverHooks: path.resolve(
+				PATH_TO_CWD,
+				svelteConfig.kit?.files?.hooks?.server || path.resolve("src", "hooks.server"),
+			),
+		}
+
+		const languageInUrl =
+			settings.languageNegotiation.strategies.some(({ type }) => type === "url") || false
+
+		const rootRoutesFolder = path.resolve(files.routes, languageInUrl ? "[lang]" : "")
+		const isStatic =
+			(await shouldContentBePrerendered(nodeishFs, files.routes)) ||
+			(await shouldContentBePrerendered(nodeishFs, rootRoutesFolder))
+
+		const usesTypeScript = await doesPathExist(
+			nodeishFs,
+			path.resolve(PATH_TO_CWD, "tsconfig.json"),
+		)
+
+		const svelteKitVersion = await getSvelteKitVersion()
+
+		resolve({
+			debug: settings.debug,
+
+			sourceLanguageTag: inlang.config().sourceLanguageTag,
+			languageTags: inlang.config().languageTags,
+			messages: () => Object.values(inlang.query.messages.getAll()),
```

Comment:
could you also use a combination of includedMessageIds and get here? We want to remove getAll. @ivanhofer

Reply:
@NilsJacobsen should be doable. 

```ts
messages: () => inlang.query.message.includedIds().map((id) => inlang.query.get({ where: id } ))
```